### PR TITLE
Update for 1.6 linking and handle stdlibs that aren't in the sysimage

### LINKED ISSUE #470: Package CompilerSupportLibraries_jll is required but does not seem to be installed
On julia nightly (1.6.0-DEV.1673 65382c708d)
```
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.6.0-DEV.1673 (2020-12-05)
 _/ |\__'_|_|_|\__'_|  |  Commit 65382c708d (0 days old master)
|__/                   |

(@v1.6) pkg> activate --temp
  Activating new environment at `/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_9X0qAU/Project.toml`

(jl_9X0qAU) pkg> dev /Users/ian/Documents/GitHub/PackageCompiler.jl
Path `/Users/ian/Documents/GitHub/PackageCompiler.jl` exists and looks like the correct package. Using existing path.
   Resolving package versions...
Updating `/private/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_9X0qAU/Project.toml`
  [9b87118b] + PackageCompiler v1.2.3 `~/Documents/GitHub/PackageCompiler.jl`
Updating `/private/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_9X0qAU/Manifest.toml`
  [9b87118b] + PackageCompiler v1.2.3 `~/Documents/GitHub/PackageCompiler.jl`
  [0dad84c5] + ArgTools
  [56f22d72] + Artifacts
  [2a0f44e3] + Base64
  [ade2ca70] + Dates
  [f43a241f] + Downloads
  [b77e0a4c] + InteractiveUtils
  [b27032c2] + LibCURL
  [76f85450] + LibGit2
  [8f399da3] + Libdl
  [56ddb016] + Logging
  [d6f4376e] + Markdown
  [ca575930] + NetworkOptions
  [44cfe95a] + Pkg
  [de0858da] + Printf
  [3fa0cd96] + REPL
  [9a3f8284] + Random
  [ea8e919c] + SHA
  [9e88b42a] + Serialization
  [6462fe0b] + Sockets
  [fa267f1f] + TOML
  [a4e569a6] + Tar
  [cf7118a7] + UUIDs
  [4ec0a83e] + Unicode
  [deac9b47] + LibCURL_jll
  [29816b5a] + LibSSH2_jll
  [c8ffd9c3] + MbedTLS_jll
  [14a3606d] + MozillaCACerts_jll
  [83775a58] + Zlib_jll
  [8e850ede] + nghttp2_jll

(jl_9X0qAU) pkg> add Flux
    Updating registry at `~/.julia/registries/General`
   Resolving package versions...
   Installed OffsetArrays ─ v1.4.1
   Installed StaticArrays ─ v1.0.0
Updating `/private/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_9X0qAU/Project.toml`
  [587475ba] + Flux v0.11.2
Updating `/private/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_9X0qAU/Manifest.toml`
  [621f4979] + AbstractFFTs v0.5.0
  [1520ce14] + AbstractTrees v0.3.3
  [79e6a3ab] + Adapt v2.3.0
  [4fba245c] + ArrayInterface v2.14.3
  [4c555306] + ArrayLayouts v0.4.12
  [ab4f0b2a] + BFloat16s v0.1.0
  [b99e7846] + BinaryProvider v0.5.10
  [fa961155] + CEnum v0.4.1
  [052768ef] + CUDA v2.3.0
  [082447d4] + ChainRules v0.7.34
  [d360d2e6] + ChainRulesCore v0.9.22
  [cdddcdb0] + ChainRulesTestUtils v0.5.4
  [944b1d66] + CodecZlib v0.7.0
  [3da002f7] + ColorTypes v0.10.9
  [5ae59095] + Colors v0.12.4
  [bbf7d656] + CommonSubexpressions v0.3.0
  [34da2185] + Compat v3.23.0
  [9a962f9c] + DataAPI v1.4.0
  [864edb3b] + DataStructures v0.18.8
  [163ba53b] + DiffResults v1.0.3
  [b552c78f] + DiffRules v1.0.2
  [ffbed154] + DocStringExtensions v0.8.3
  [e2ba6199] + ExprTools v0.1.3
  [1a297f60] + FillArrays v0.10.1
  [26cc04aa] + FiniteDifferences v0.11.5
  [53c48c17] + FixedPointNumbers v0.8.4
  [587475ba] + Flux v0.11.2
  [f6369f11] + ForwardDiff v0.10.14
  [d9f16b24] + Functors v0.1.0
  [0c68f7d7] + GPUArrays v6.1.2
  [61eb1bfa] + GPUCompiler v0.8.3
  [0e44f5e4] + Hwloc v1.0.3
  [7869d1d1] + IRTools v0.4.2
  [615f187c] + IfElse v0.1.0
  [692b3bcd] + JLLWrappers v1.1.3
  [e5e0dc1b] + Juno v0.8.4
  [929cbde3] + LLVM v3.4.0
  [bdcacae8] + LoopVectorization v0.9.1
  [1914dd2f] + MacroTools v0.5.6
  [e89f7d12] + Media v0.5.0
  [e1d29d7a] + Missings v0.4.4
  [46d2c3a1] + MuladdMacro v0.2.2
  [872c559c] + NNlib v0.7.7
  [77ba4419] + NaNMath v0.3.5
  [6fe1bfb0] + OffsetArrays v1.4.1
  [bac558e1] + OrderedCollections v1.3.2
  [189a3867] + Reexport v0.2.0
  [ae029012] + Requires v1.1.1
  [708f8203] + Richardson v1.2.0
  [476501e8] + SLEEFPirates v0.6.0
  [6c6a2e73] + Scratch v1.0.3
  [a2af1166] + SortingAlgorithms v0.3.1
  [276daf66] + SpecialFunctions v1.1.0
  [90137ffa] + StaticArrays v1.0.0
  [2913bbd2] + StatsBase v0.33.2
  [a759f4b9] + TimerOutputs v0.5.7
  [3bb67fe8] + TranscodingStreams v0.9.5
  [3a884ed6] + UnPack v1.0.2
  [3d5dd08c] + VectorizationBase v0.13.6
  [a5390f91] + ZipFile v0.9.3
  [e88e6eb3] + Zygote v0.5.15
  [700de1a5] + ZygoteRules v0.2.0
  [efe28fd5] + OpenSpecFun_jll v0.5.3+4
  [8bb1440f] + DelimitedFiles
  [8ba89e20] + Distributed
  [37e2e46d] + LinearAlgebra
  [a63ad114] + Mmap
  [9abbd945] + Profile
  [1a1011a3] + SharedArrays
  [2f01184e] + SparseArrays
  [10745b16] + Statistics
  [8dfed614] + Test
  [e66e0078] + CompilerSupportLibraries_jll
  Progress [========================================>]  67/67
67 dependencies successfully precompiled in 53 seconds (1 already precompiled)

julia> using PackageCompiler

julia> create_sysimage(:Flux; sysimage_path="FluxSysimage.so")
[ Info: PackageCompiler: creating system image object file, this might take a while...
ERROR: LoadError: LoadError: LoadError: LoadError: LoadError: LoadError: LoadError: ArgumentError: Package CompilerSupportLibraries_jll [e66e0078-7015-5450-92f7-15fbd957f2ae] is required but does not seem to be installed:
 - Run `Pkg.instantiate()` to install all recorded dependencies.

Stacktrace:
  [1] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:985
  [2] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:909
  [3] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:896
  [4] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
  [5] top-level scope
    @ ~/.julia/packages/JLLWrappers/KuIwt/src/toplevel_generators.jl:170
  [6] include
    @ ./Base.jl:386 [inlined]
  [7] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1045
  [8] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:909
  [9] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:896
 [10] include
    @ ./Base.jl:386 [inlined]
 [11] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1045
 [12] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:909
 [13] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:896
 [14] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
 [15] include(x::String)
    @ Zygote.Forward ~/.julia/packages/Zygote/nK6sg/src/forward/Forward.jl:1
 [16] top-level scope
    @ ~/.julia/packages/Zygote/nK6sg/src/forward/Forward.jl:11
 [17] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
 [18] include(x::String)
    @ Zygote ~/.julia/packages/Zygote/nK6sg/src/Zygote.jl:1
 [19] top-level scope
    @ ~/.julia/packages/Zygote/nK6sg/src/Zygote.jl:22
 [20] include
    @ ./Base.jl:386 [inlined]
 [21] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1045
 [22] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:909
 [23] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:896
 [24] include
    @ ./Base.jl:386 [inlined]
 [25] _require(pkg::Base.PkgId)
    @ Base ./loading.jl:1045
 [26] require(uuidkey::Base.PkgId)
    @ Base ./loading.jl:909
 [27] require(into::Module, mod::Symbol)
    @ Base ./loading.jl:896
in expression starting at /Users/ian/.julia/packages/OpenSpecFun_jll/Xw8XK/src/wrappers/x86_64-apple-darwin-libgfortran5.jl:4
in expression starting at /Users/ian/.julia/packages/OpenSpecFun_jll/Xw8XK/src/OpenSpecFun_jll.jl:2
in expression starting at /Users/ian/.julia/packages/SpecialFunctions/24S26/src/SpecialFunctions.jl:1
in expression starting at /Users/ian/.julia/packages/Zygote/nK6sg/src/forward/number.jl:1
in expression starting at /Users/ian/.julia/packages/Zygote/nK6sg/src/forward/Forward.jl:1
in expression starting at /Users/ian/.julia/packages/Zygote/nK6sg/src/Zygote.jl:1
in expression starting at /Users/ian/.julia/packages/Flux/q3zeA/src/Flux.jl:1
ERROR: failed process: Process(`/Users/ian/Documents/GitHub/julia/usr/bin/julia --color=yes --startup-file=no --cpu-target=native --sysimage=/Users/ian/Documents/GitHub/julia/usr/lib/julia/sys.dylib --project=/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_9X0qAU --output-o=/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_Rgv1P3.o -e 'Base.reinit_stdio()
@eval Sys BINDIR = ccall(:jl_get_julia_bindir, Any, ())::String
Base.init_load_path()
if isdefined(Base, :init_active_project)
    Base.init_active_project()
end
Base.init_depot_path()
import Flux
# This @eval prevents symbols from being put into Main
@eval Module() begin
    PrecompileStagingArea = Module()
    for (_pkgid, _mod) in Base.loaded_modules
        if !(_pkgid.name in ("Main", "Core", "Base"))
            eval(PrecompileStagingArea, :(const $(Symbol(_mod)) = $_mod))
        end
    end
    precompile_statements = String[]
        append!(precompile_statements, readlines("/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_gAxcfO"))

    for statement in sort(precompile_statements)
        # println(statement)
        # The compiler has problem caching signatures with \`Vararg{?, N}\`. Replacing
        # N with a large number seems to work around it.
        statement = replace(statement, r"Vararg{(.*?), N} where N" => s"Vararg{\1, 100}")
        try
            Base.include_string(PrecompileStagingArea, statement)
        catch
            # See julia issue #28808
            @debug "failed to execute $statement"
        end
    end
end # module
empty!(LOAD_PATH)
empty!(DEPOT_PATH)
'`, ProcessExited(1)) [1]

Stacktrace:
 [1] pipeline_error
   @ ./process.jl:525 [inlined]
 [2] run(::Cmd; wait::Bool)
   @ Base ./process.jl:440
 [3] run
   @ ./process.jl:438 [inlined]
 [4] create_sysimg_object_file(object_file::String, packages::Vector{String}; project::String, base_sysimage::String, precompile_execution_file::Vector{String}, precompile_statements_file::Vector{String}, cpu_target::String, script::Nothing, isapp::Bool)
   @ PackageCompiler ~/Documents/GitHub/PackageCompiler.jl/src/PackageCompiler.jl:298
 [5] create_sysimage(packages::Symbol; sysimage_path::String, project::String, precompile_execution_file::Vector{String}, precompile_statements_file::Vector{String}, incremental::Bool, filter_stdlibs::Bool, replace_default::Bool, cpu_target::String, script::Nothing, base_sysimage::Nothing, isapp::Bool)
   @ PackageCompiler ~/Documents/GitHub/PackageCompiler.jl/src/PackageCompiler.jl:429
 [6] top-level scope
   @ REPL[7]:1

```

`CompilerSupportLibraries_jll` is declared in `OpenSpecFun_jll`
https://github.com/JuliaBinaryWrappers/OpenSpecFun_jll.jl/blob/6c505cce3bdcd9cd2b15b4f9362ec3a42c4da71c/Project.toml#L8

and `pkg> instantiate` doesn't fix this.

### PR DESCRIPTION
**Pull request by ianshmean**
Doesn't work yet. Attempts to handle stdlibs that aren't in the sysimage 

Trying to fix #470 

@KristofferC I thought it might be helpful to start this. Do you see where the remaining work is needed?

### CODE REVIEW

**Comment by ianshmean** [comment]
I thought it may just need this
https://github.com/JuliaLang/PackageCompiler.jl/blob/942c5a607001ca639517f4ef92e6caa2aa201c87/src/PackageCompiler.jl#L219

changed to 
```julia
for (_pkgid, _mod) in Base._sysimage_modules
```

but that doesn't do the trick

**Comment by ianshmean** [comment]
I don't quite understand why this is failing. I added some status prints during the `julia_code`.

Surely if the packages were truly not installed they'd show up as such in the `status` reports?
Also, `@stdlib` is in the `LOAD_PATH` so it should be able to find `CompilerSupportLibraries_jll`

```
julia> create_sysimage(:OpenSpecFun_jll; sysimage_path="Sysimage.so")
  Progress [========================================>]  1/1
1 dependency successfully precompiled in 1 seconds (6 already precompiled)
[ Info: PackageCompiler: creating system image object file, this might take a while...
Status `/private/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_FVUhJp/Project.toml`
  [9b87118b] PackageCompiler v1.2.3 `~/Documents/GitHub/PackageCompiler.jl`
  [efe28fd5] OpenSpecFun_jll v0.5.3+4
Status `/private/var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_FVUhJp/Manifest.toml`
  [692b3bcd] JLLWrappers v1.1.3
  [9b87118b] PackageCompiler v1.2.3 `~/Documents/GitHub/PackageCompiler.jl`
  [efe28fd5] OpenSpecFun_jll v0.5.3+4
  [0dad84c5] ArgTools
  [56f22d72] Artifacts
  [2a0f44e3] Base64
  [ade2ca70] Dates
  [f43a241f] Downloads
  [b77e0a4c] InteractiveUtils
  [b27032c2] LibCURL
  [76f85450] LibGit2
  [8f399da3] Libdl
  [56ddb016] Logging
  [d6f4376e] Markdown
  [ca575930] NetworkOptions
  [44cfe95a] Pkg
  [de0858da] Printf
  [3fa0cd96] REPL
  [9a3f8284] Random
  [ea8e919c] SHA
  [9e88b42a] Serialization
  [6462fe0b] Sockets
  [fa267f1f] TOML
  [a4e569a6] Tar
  [cf7118a7] UUIDs
  [4ec0a83e] Unicode
  [e66e0078] CompilerSupportLibraries_jll
  [deac9b47] LibCURL_jll
  [29816b5a] LibSSH2_jll
  [c8ffd9c3] MbedTLS_jll
  [14a3606d] MozillaCACerts_jll
  [83775a58] Zlib_jll
  [8e850ede] nghttp2_jll
Base.LOAD_PATH = ["@", "@v#.#", "@stdlib"]
ERROR: LoadError (x2): ArgumentError: Package CompilerSupportLibraries_jll [e66e0078-7015-5450-92f7-15fbd957f2ae] is required but does not seem to be installed:
 - Run `Pkg.instantiate()` to install all recorded dependencies.
```

**Comment by ianshmean** [comment]
It was failing to find `CompilerSupportLibraries_jll` because `Sys.STDLIB` was `== ""`, perhaps due to 
https://github.com/JuliaLang/julia/blob/65382c708d1fafeebb7b896dfeb2795362fe8921/base/sysimg.jl#L132

After restoring that, I get further:
```
julia> create_sysimage(:OpenSpecFun_jll; sysimage_path="Sysimage.so")
  Progress [========================================>]  1/1
1 dependency successfully precompiled in 1 seconds (6 already precompiled)
[ Info: PackageCompiler: creating system image object file, this might take a while...
Undefined symbols for architecture x86_64:
  "_jl_RTLD_DEFAULT_handle", referenced from:
      _julia_typeinf_ext_toplevel_51330 in jl_cCT22h.o(text.o)
      _julia_typeinf_ext_toplevel_12632 in jl_cCT22h.o(text.o)
      _japi1_get_type_call_53289 in jl_cCT22h.o(text.o)
      _julia_coverage_enabled_13029 in jl_cCT22h.o(text.o)
      _julia_inbounds_option_11479 in jl_cCT22h.o(text.o)
      _julia_NativeInterpreter_7825 in jl_cCT22h.o(text.o)
      _julia_verify_linetable_13177 in jl_cCT22h.o(text.o)
      ...
...
  "_jl_world_counter", referenced from:
      _jlcapi_cfunction_45704 in jl_cCT22h.o(text.o)
      _jlcapi_cfunction_45737 in jl_cCT22h.o(text.o)
      _jlcapi_cfunction_45772 in jl_cCT22h.o(text.o)
      _jlcapi_timeout_callback_52506 in jl_cCT22h.o(text.o)
      _jlcapi_cleanup_callback_52463 in jl_cCT22h.o(text.o)
      _jlcapi_timer_callback_52376 in jl_cCT22h.o(text.o)
      _jlcapi_socket_callback_52380 in jl_cCT22h.o(text.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ERROR: failed process: Process(`gcc -m64 -march=x86-64 -shared -L/Users/ian/Documents/GitHub/julia/usr/lib -o Sysimage.so -Wl,-all_load /var/folders/_6/1yf6sj0950vcg4t91m9ltb5w0000gn/T/jl_cCT22h.o -ljulia`, ProcessExited(1)) [1]
```

File: src/PackageCompiler.jl [hunk 1]
```diff
@@ -239,11 +242,19 @@ function create_sysimg_object_file(object_file::String, packages::Vector{String}
...
             Base.init_active_project()
         end
         Base.init_depot_path()
+
+        # TODO: Remove before merge
+        import Pkg
+        Pkg.status(mode = Pkg.PKGMODE_PROJECT)
+        Pkg.status(mode = Pkg.PKGMODE_MANIFEST)
+        @show Base.LOAD_PATH
+        @show Base.load_path()
```
Comment (ianshmean) [inline src/PackageCompiler.jl:257]: Just debug code

**Comment by ianshmean** [comment]
The issue I'm seeing locally is replicated on MacOS Nightly CI https://github.com/JuliaLang/PackageCompiler.jl/pull/471/checks?check_run_id=1505423082#step:4:156

**Comment by ianshmean** [comment]
Seems like various additional issues going on too

Linux with 1.5.3 https://github.com/JuliaLang/PackageCompiler.jl/pull/471/checks?check_run_id=1505559609#step:4:312
```
 Info: PackageCompiler: creating base system image (incremental=false)...
FATAL ERROR: Symbol "ccall_jl_n_threads_39785"not found
signal (6): Aborted
in expression starting at none:0
gsignal at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
abort at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
addModule at /buildworker/worker/package_linux64/build/src/jitlayers.cpp:640
jl_add_to_ee at /buildworker/worker/package_linux64/build/src/jitlayers.cpp:893 [inlined]
jl_add_to_ee at /buildworker/worker/package_linux64/build/src/jitlayers.cpp:955
jl_add_to_ee at /buildworker/worker/package_linux64/build/src/jitlayers.cpp:940
jl_add_to_ee at /buildworker/worker/package_linux64/build/src/jitlayers.cpp:977 [inlined]
```

MacOS with 1.5.3 https://github.com/JuliaLang/PackageCompiler.jl/pull/471/checks?check_run_id=1505559615#step:4:157
```
FATAL ERROR: Symbol "ccall_jl_safe_printf_41829"not found
signal (6): Abort trap: 6
in expression starting at none:0
__pthread_kill at /usr/lib/system/libsystem_kernel.dylib (unknown line)
Allocations: 87348652 (Pool: 87323432; Big: 25220); GC: 96
```

Windows with 1.5.3 https://github.com/JuliaLang/PackageCompiler.jl/pull/471/checks?check_run_id=1505559620#step:4:74
```
Exception: EXCEPTION_ACCESS_VIOLATION at 0x6911a168 -- uv_write at /workspace/srcdir/libuv\src/win\stream.c:124
in expression starting at none:0
uv_write at /workspace/srcdir/libuv\src/win\stream.c:124
jl_uv_write at /cygdrive/c/buildbot/worker/package_win64/build/src\jl_uv.c:463
uv_write_async at .\stream.jl:965
uv_write at .\stream.jl:922
unsafe_write at .\stream.jl:1005
write at .\strings\io.jl:183 [inlined]
print at .\strings\io.jl:185 [inlined]
#with_output_color#745 at .\util.jl:76
unknown function (ip: 000000001c693c74)
```

File: src/PackageCompiler.jl [hunk 2]
```diff
@@ -239,11 +249,19 @@ function create_sysimg_object_file(object_file::String, packages::Vector{String}
     julia_code = """
         Base.reinit_stdio()
         @eval Sys BINDIR = ccall(:jl_get_julia_bindir, Any, ())::String
+        @eval Sys STDLIB = "$(Sys.BINDIR)/../share/julia/stdlib/v$(VERSION.major).$(VERSION.minor)"
```
Comment (KristofferC) [inline src/PackageCompiler.jl:252]: This should be enough to fix it, or? I don't see why being in the sysimage or not should matter.

Comment (ianshmean) [reply src/PackageCompiler.jl:250]: Yeah, this is the main fix. The other stuff just prevents all of the new jll's being added by default when a new sysimage is created. They would've all been added, now they're only added if used

**Comment by ianshmean** [comment]
Some progress on nightly. I think I've fixed https://github.com/JuliaLang/julia/issues/38749

But the HelloWorldC_jll artifact is having issues in the penultimate tests.

**Comment by KristofferC** [comment]
> But the HelloWorldC_jll artifact 

I think that is due to an Artifact change, I remember having that error quite a long time ago (https://github.com/JuliaLang/PackageCompiler.jl/pull/460#issuecomment-713342465)

**Comment by ianshmean** [comment]
I see ok. 

Feel free to push to this. I won't have time to work on it for a day or so

**Comment by ianshmean** [comment]
Just confirmed that this works in the wild on nightly on linux with a pretty big manifest

File: src/PackageCompiler.jl (hunk 2 above)
Comment (visr) [reply src/PackageCompiler.jl:250]: On Windows this ends up as a syntax error:
```julia
julia> "c:\bin\julia-1.6\bin/../share/julia/stdlib/v1.6"
ERROR: syntax: invalid escape sequence
```

With `repr` we can work around that, for instance like this:
```julia
@eval Sys STDLIB = $(repr(abspath(Sys.BINDIR, "../share/julia/stdlib", string('v', VERSION.major, '.', VERSION.minor))))
```

Comment (ianshmean) [reply src/PackageCompiler.jl]: Ah yes, thanks

**Comment by KristofferC** [comment]
Cool, let's merge this and get a release out as an improvement and I'll look at the artifact tests at some point.

**Pull request merged**